File: data/java/tests/test_Pydantic_constr.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_Pydantic_constr {

    @Test
    public void test_valid_profile() {
        function_Pydantic_constr handler = new function_Pydantic_constr();
        String result = handler.create_user_profile("user_123", "user@example.com");
        assertEquals("username='user_123' email='user@example.com'", result);
    }

    @Test
    public void test_invalid_email() {
        function_Pydantic_constr handler = new function_Pydantic_constr();
        String result = handler.create_user_profile("user123", "user.com");
        assertTrue(result.contains("value is not a valid email address"));
    }

    @Test
    public void test_valid_profile_with_min_length_username() {
        function_Pydantic_constr handler = new function_Pydantic_constr();
        String result = handler.create_user_profile("abc", "test@example.com");
        assertEquals("username='abc' email='test@example.com'", result);
    }

    @Test
    public void test_valid_profile_with_max_length_username() {
        function_Pydantic_constr handler = new function_Pydantic_constr();
        String result = handler.create_user_profile("aaaaaaaaaaaaaaaaaaaa", "longusername@example.com");
        assertEquals("username='aaaaaaaaaaaaaaaaaaaa' email='longusername@example.com'", result);
    }

    @Test
    public void test_invalid_email_missing_domain() {
        function_Pydantic_constr handler = new function_Pydantic_constr();
        String result = handler.create_user_profile("user123", "user@");
        assertTrue(result.contains("value is not a valid email address"));
    }
}

File: data/python/tests/test_Pydantic_constr.py
import unittest
from function_Pydantic_constr import UserProfileHandler

class TestCreateUserProfile(unittest.TestCase):
    def test_valid_profile(self):
        handler = UserProfileHandler()
        result = handler.create_user_profile("user_123", "user@example.com")
        self.assertEqual(result, "username='user_123' email='user@example.com'")

    def test_invalid_email(self):
        handler = UserProfileHandler()
        result = handler.create_user_profile("user123", "user.com")
        self.assertIsInstance(result, str)
        self.assertIn('value is not a valid email address', result)

    def test_valid_profile_with_min_length_username(self):
        handler = UserProfileHandler()
        result = handler.create_user_profile("abc", "test@example.com")
        self.assertEqual(result, "username='abc' email='test@example.com'")

    def test_valid_profile_with_max_length_username(self):
        handler = UserProfileHandler()
        result = handler.create_user_profile("a" * 20, "longusername@example.com")
        self.assertEqual(result, "username='aaaaaaaaaaaaaaaaaaaa' email='longusername@example.com'")

    def test_invalid_email_missing_domain(self):
        handler = UserProfileHandler()
        result = handler.create_user_profile("user123", "user@")
        self.assertIsInstance(result, str)
        self.assertIn('value is not a valid email address', result)

if __name__ == "__main__":
    unittest.main()
